### PR TITLE
Hub

### PR DESCRIPTION
**Pull request by vesteinn**

### CODE REVIEW

File: src/greynirseq/nicenlp/examples/pos/README.md [hunk 1]
```diff
@@ -1,14 +1,43 @@
...
+
+### Using torch hub
+
+This will download the model from our servers and return an instance for inference
+
+```python
+import torch
+model = torch.hub.load("mideind/GreynirSeq", "icebert.pos")
+model.eval()
+labels = model.predict_labels(["Systurnar Guðrún og Monique átu einar um jólin á McDonalds ."]
```
Comment (peturorri) [inline src/greynirseq/nicenlp/examples/pos/README.md:21]: Vantar að loka sviga.

File: src/greynirseq/cli/greynirseq.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    )
+
+    args = parser.parse_args()
+    if args.command.lower() == "ner":
+        command = NER
+
+    elif args.command.lower() == "pos":
+        command = POS
+
+    handler = command(args.input, args.output, args.device, args.bsz, args.progress, args.max_input_words_split)
```
Comment (peturorri) [inline src/greynirseq/cli/greynirseq.py:130]: Er ekki hugsað fyrir því að ákveðin módel eða verkefni geti bætt við sínum eigin flöggum á svipaðan hátt og fairseq gerir?

File: src/greynirseq/cli/greynirseq.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        raise NotImplementedError
+
+    def handle_batch(self, batch):
+        results = self.infer(batch)
+        for result in results:
+            self.output.write(result + "\n")
+
+    def run(self):
+        batch = []
+        input_iterable = map(str.rstrip, self.input)
```
Comment (peturorri) [inline src/greynirseq/cli/greynirseq.py:33]: Af hverju rstrip en ekki strip? Hvað erum við að fjarlægja hér?

File: src/greynirseq/cli/greynirseq.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+    args = parser.parse_args()
+    if args.command.lower() == "ner":
+        command = NER
+
+    elif args.command.lower() == "pos":
+        command = POS
+
+    handler = command(args.input, args.output, args.device, args.bsz, args.progress, args.max_input_words_split)
+    handler.run()
```
Comment (peturorri) [inline src/greynirseq/cli/greynirseq.py:131]: Mér finnst eitthvað skrítið við að kalla í run með engum argumentum. En það er kannski allt í lagi að hafa input sem constuctor argument.

**Review by peturorri — COMMENTED** [review]
@haukurb lenti í því að verkið hans feilaði þegar serverinn sem hýsir módelin fór niður. Sé ekki augljóslega hér hvers vegna það gerist en það er frekar glatað.

File: src/greynirseq/cli/greynirseq.py (hunk 3 above)
Comment (vesteinn) [reply src/greynirseq/cli/greynirseq.py:33]: Aðalega newline

File: src/greynirseq/cli/greynirseq.py (hunk 2 above)
Comment (vesteinn) [reply src/greynirseq/cli/greynirseq.py:130]: Þá þarf bara aðeins að bæta því við þegar það gerist, fínt að nota subparser til þess. En ekki þörf á því núna, hálfgert MVP

File: src/greynirseq/cli/greynirseq.py (hunk 4 above)
Comment (vesteinn) [reply src/greynirseq/cli/greynirseq.py:131]: Færi input og output í run

File: README.md [hunk 5]
```diff
@@ -6,52 +10,71 @@ GreynirSeq is a natural language parsing toolkit for Icelandic focused on sequen
...
-IceBERT is an Icelandic language model.
+### TL;DR give me the CLI
+
+The `greynirseq` CLI interface can be used to run state-of-the-art POS and NER tagging for Icelandic. Run `pip install greynirseq && greynirseq -h` to see what options are available.
+
+#### POS
+
+``` bash
+❯ pip install greynirseq
+❯ echo "Systurnar Guðrún og Monique átu einar um jólin á McDonalds ." | greynirseq pos --input -
```
Comment (HaukurPall) [inline README.md:37]: Kannski líka bæta við dæmi sem sýnir að cli styður líka skrár.

Comment (HaukurPall) [inline README.md:37]: Við skulum líka nefna það með berum orðum að við gerum ráð fyrir að inntak sé tilreitt.

File: src/greynirseq/nicenlp/examples/pos/README.md [hunk 6]
```diff
@@ -1,15 +1,28 @@
...
 
-This example shows how to train an Icelandic POS tagger with ~98% accuracy on the [Tagged Icelandic Corpus](http://www.malfong.is/index.php?lang=en&pg=mim) (MIM) dataset.
+This example shows how to train an Icelandic POS tagger with ~98.2% accuracy on the [Tagged Icelandic Corpus](http://www.malfong.is/index.php?lang=en&pg=mim) (MIM) dataset . The output can be configure to use the MIM 2.0 label format which is the default for the CLI.
 
 ## Preprocessing
 See `./prep_mim_pos.sh` which is setup to process all data from the MIM pos set and prepare for crossvalidation.
 
 ## Training
-See `./train.sh` which trains all ten sets for crossvalidation.
+See `./train.sh` which trains all ten sets for cross-validation.
```
Comment (HaukurPall) [inline src/greynirseq/nicenlp/examples/pos/README.md:9]: `train.sh` fyrir POS virðist ennþá vera óyfirfarið.

File: pyproject.toml [hunk 7]
```diff
@@ -15,12 +15,42 @@ testpaths = [
 name = "greynirseq"
 version = "0.1"
 description = "Natural language processing for Icelandic"
-authors = []
+authors = ["Miðeind ehf <tauganet@mideind.is>"]
```
Comment (HaukurPall) [inline pyproject.toml:18]: Við erum með setup.py og pyproject.toml sem innihalda mismunandi upplýsingar um pakkann.
pyproject skilgreinir package dependencies en setup.py skilgreinir hvernig á að byggja cython dependencies.

Virka þessar tvær skrár saman?

**Review by HaukurPall — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 7 above)
Comment (HaukurPall) [reply pyproject.toml:18]: Við getum reynt: https://github.com/mdgoldberg/poetry-cython-example

Comment (vesteinn) [reply pyproject.toml:18]: Let's take that as a part of the constituency parsing

File: src/greynirseq/cli/greynirseq.py [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+
+import argparse
+import sys
+
+import torch
+import tqdm
+
+
+class GreynirSeqIO:
+    def __init__(self, input, output, device, batch_size, show_progress, max_input_words_split):
```
[the remark this answers is not in the document]
Comment (vesteinn) [reply src/greynirseq/cli/greynirseq.py]: They are there

File: README.md (hunk 5 above)
Comment (vesteinn) [reply README.md:37]: This is the TL;DR, `greynirseq -h` is mentioned above

File: src/greynirseq/cli/greynirseq.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (vesteinn) [reply src/greynirseq/cli/greynirseq.py]: https://github.com/mideind/GreynirSeq/pull/26/files

**Review by HaukurPall — APPROVED** [review]
